### ISSUE TITLE
Node hangs on Windows 10 when executing the example code

### ISSUE THREAD

**Issue by SteveShe** [tediousjs/tedious#439]
I have been trying for several hours to get to the bottom of this and I just can't get there. Whenever I run a script that uses tedious, it does everything it is supposed to do, but node never returns to the command prompt. For completeness I will give you the entire command line session and the code I executed to reproduce this problem:

```
C:\Users\steveshe\dev
λ md test

C:\Users\steveshe\dev
λ cd test

C:\Users\steveshe\dev\test
λ npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help json` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg> --save` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
name: (test)
version: (1.0.0)
description:
entry point: (index.js)
test command:
git repository:
keywords:
author:
license: (ISC)
About to write to C:\Users\steveshe\dev\test\package.json:

{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}


Is this ok? (yes)

C:\Users\steveshe\dev\test  (test@1.0.0)
λ npm i tedious
npm WARN package.json test@1.0.0 No description
npm WARN package.json test@1.0.0 No repository field.
npm WARN package.json test@1.0.0 No README data
tedious@1.14.0 node_modules\tedious
├── sprintf@0.1.5
├── semver@5.3.0
├── big-number@0.3.1
├── iconv-lite@0.4.13
├── readable-stream@2.1.5 (buffer-shims@1.0.0, string_decoder@0.10.31, inherits@2.0.1, process-nextick-args@1.0.7, util-deprecate@1.0.2, core-util-is@1.0.2, isarray@1.0.0)
├── bl@1.1.2 (readable-stream@2.0.6)
└── babel-runtime@5.8.38 (core-js@1.2.7)

C:\Users\steveshe\dev\test  (test@1.0.0)
λ notepad app.js

C:\Users\steveshe\dev\test  (test@1.0.0)
λ node app.js
debugtext: connected to server.database.windows.net:1433
debugtext: State change: Connecting -> SentPrelogin
debugtext: State change: SentPrelogin -> SentTLSSSLNegotiation
debugtext: TLS negotiated (ECDHE-RSA-AES256-SHA384, TLSv1/SSLv3)
debugtext: State change: SentTLSSSLNegotiation -> SentLogin7WithStandardLogin
debugtext: Packet size changed from 4096 to 4096
debugtext: State change: SentLogin7WithStandardLogin -> LoggedInSendingInitialSql
debugtext: State change: LoggedInSendingInitialSql -> LoggedIn
debugtext: State change: LoggedIn -> SentClientRequest
42
hello world
debugtext: State change: SentClientRequest -> LoggedIn
1 rows
^C
C:\Users\steveshe\dev\test  (test@1.0.0)
λ 
```

From this point onward the command prompt just sits here without returning. A ctrl+c will kill node, but that is really a sub-optimal solution. FWIW, same thing happens in a regular Windows CMD prompt and when running it under in VSCode. 

Here is the code in the app.js file:

```
var Connection = require('tedious').Connection;

var config = {  
    userName: 'user',  
    password: 'password',  
    server: 'server.database.windows.net',  
    options: {encrypt: true, database: 'db001'}  
}

var connection = new Connection(config);
connection.on('error', function(err) {  
    if (err) {  
        console.log(err);
    }else {  
        console.log("Error called with no err object.");
    }  
});
connection.on('debug', function(messagetext) {  
    if (messagetext) {  
        console.log('debugtext: ' + messagetext);
    }else {  
        console.log("Debug called with no messagetext object.");
    }  
});  
connection.on('connect', function(err) {
// If no error, then good to go...
    executeStatement();
}
);

var Request = require('tedious').Request;

  function executeStatement() {
    request = new Request("select 42, 'hello world'", function(err, rowCount) {
      if (err) {
        console.log(err);
      } else {
        console.log(rowCount + ' rows');
      }
    });

    request.on('row', function(columns) {
      columns.forEach(function(column) {
        console.log(column.value);
      });
    });

    connection.execSql(request);
  }
```

**Comment by arthurschreiber** [comment]
Hey @SteveShe,

I went through your comment and re-formatted it to be a bit more readable, I hope that's ok! 😄 

Node seems to be "hanging" because the connection to the sql server is still open. I put that in quotation marks because that's the expected behaviour - whenever you still have a something that keeps the event loop of Node.JS alive, the process won't shut down but will keep running.

In your case, the easiest option would be to close the connection by calling `connection.close()` once you're done using it, e.g. inside the callback that you pass to `new Request`.

I hope this is helpful information!

As this is not a bug in tedious, I will close this issue. 👍

**Comment by SteveShe** [comment]
Thanks Arthur. I tried that and thought it didn't work. I suspect that is because early on I validated that I got the  same result sunder VSCode as I did from the command line and must have assumed that would always be the case. As it strands now, when I run it from the command line with the connection close(), it exits as expected. However, when I run it under VSCode it does not. I'll open a bug there just for the sake of completeness.
